platform-plugins.md

```diff
@@ -0,0 +1,277 @@
+
+**Step 1: Create the Flutter platform client**
+
+The Flutter platform client code is written inside the app's `State` class.
+Start by constructing the channel. Since we need just a single platform message,
+`getBatteryLevel`, representing a platform method call, we will use a
+`PlatformMethodChannel`:
+
+```dart
+class _PlatformServicesState extends State<PlatformServices> {
+  static const PlatformMethodChannel platform = const PlatformMethodChannel('battery');
+  ...
+}
+```
+
+The client and host sides of the channel are connected through by the passed
+name argument; here `battery`. Note that channel names must be unique; we
+recommend using a naming pattern using *TODO*.
+
+Next, we invoke a method on the method channel, specifying the concrete method
+to call via a String `method` argument, and use the returned result to update
+our user interface:
+
+```dart
+String _batteryLevel = 'Unknown battery level.';
+
+Future<Null> _getBatteryLevel() async {
+  final int result = await platform.invokeMethod('getBatteryLevel');
+  batteryLevel = 'Battery level at $result % .';
+
+  setState(() {
+    _batteryLevel = batteryLevel;
+  });
+}
+```
+
+The call may fail -- for example if the platform does not support the platform
+API (such as when running in a simulator), so next we add exception handling:
+
+```dart
+  try {
+    final int result = await platform.invokeMethod('getBatteryLevel');
+    batteryLevel = 'Battery level at $result % .';
+  } on PlatformException catch (e) {
+    batteryLevel = "Failed to get battery level: '${e.message}'.";
+  }
+```
+
+**Step 2a: Add an iOS platform-specific implementation**
+
+*TODO*
+
+**Step 2b: Add an Android platform-specific implementation**
+
+Start by opening the Android host portion of your Flutter app in Android Studio:
+
+1. Start Android Studio
+
+1. Invoke File > Open...
+
+1. Navigate to the directory holding your Flutter app, and select the `android`
+folder inside it. Click OK.
+
+1. Open the file `MainActivity.java` located in the `java` folder in the Project
+view.
+
+Next, create a `FlutterMethodChannel` and set a `MethodCallHandler` inside the
+`onCreate` method. Make sure to use the same channel name as was used on the
+Flutter client side.
+
+```java
+@Override
+public void onCreate(Bundle savedInstanceState) {
+  private static final String CHANNEL = "battery";
+
+  super.onCreate(savedInstanceState);
+
+  new FlutterMethodChannel(getFlutterView(), CHANNEL).setMethodCallHandler(
+      new MethodCallHandler() {
+        @Override
+        public void onMethodCall(MethodCall call, Response response) {
+          // TODO
+        }
+  });
+}
+```
+
+Next, we add the actual Android Java code that uses the Android battery APIs to
+retrieve the battery level. This code is exactly the same you would have written
+in a native Android app:
+
+```java
+private int getBatteryLevel() {
```

Comment:
As with above, providing a marker for where this code goes will help our users that like to "copy and paste" their way to success. :)